code/modules/mob/mob_helpers.dm

```diff
@@ -22,6 +22,27 @@
 			return 0
 	return 1
 
+/mob/proc/get_screen_colour()
+
+/mob/dead/observer/get_screen_colour()
+	return list()
+
+/mob/proc/update_client_colour(var/time = 10) //Update the mob's client.color if the colour matrix in the eyes (or whatever) is different.
+	if(!client)
+		return
+	client.colour_transition(get_screen_colour(), time = time)
+
+/mob/living/carbon/human/get_screen_colour() //Fetch the colour matrix from wherever (e.g. eyes) so it can be compared to client.color.
+	. = ..()
+	if(.)
+		return .
+	var/obj/item/clothing/glasses/worn_glasses = glasses
+	var/obj/item/organ/internal/eyes/eyes = get_int_organ(/obj/item/organ/internal/eyes)
+	if(istype(worn_glasses) && worn_glasses.color_view) //Check to see if they got those magic glasses and they're augmenting the colour of what the wearer sees. If they're not, color_view should be null.
+		return worn_glasses.color_view
+	else if(eyes && eyes.colourmatrix && !(eyes.robotic)) //The robotic eyes check is a redundant backup to the setting of colourmatrix during robotize() in code\modules\surgery\organs\organ_internal.dm.

```

Comment:
You can drop the `robotic` sanity check here. Too many sanity checks just hide errors.

Then, make sure a Taj/Vulp with assisted/mechanical eyes and the "unaugmented" option enabled works as expected, because things might not be getting applied in the expected order.